crates/uv/src/lib.rs

```diff
@@ -42,8 +44,45 @@ pub(crate) mod printer;
 pub(crate) mod settings;
 pub(crate) mod version;
 
+/// Resolves the script target for a run command.
+/// If it's a local file, does nothing. If it's a URL, downloads the content
+/// to a temporary file and updates the command. Prioritizes local files over URLs.
+/// Returns Some(NamedTempFile) if a remote script was downloaded, None otherwise.
+async fn resolve_script_target(
+    command: &mut ExternalCommand,
+) -> Result<Option<tempfile::NamedTempFile>> {
+    let Some(target) = command.get_mut(0) else {
+        return Ok(None);
+    };
```

Comment:
What happens if there is more than one positional argument? I think this means we ignore it? Should we return `Ok(None)` instead? (I'm actually not quite sure about this one.)

Reply:
> What happens if there is more than one positional argument?

if there is more than one, we don't change them (so they are forwarded down to the python program). E.g., it would be nice to support:

```sh
uv run <url> [python program flags]
```

Reply:
Perfect. SGTM.